src/Types/Conversions/rock/Pointcloud.hpp

```diff
@@ -23,6 +43,9 @@ namespace RockConversion {
         // std::vector<base::Vector4d> colors;
         counter = 0;
         if (rock_type.colors.size()) {
+            // TODO: is this really how this should work?
```

Comment:
How should this and the following TODO comment be resolved?

- [ ] Channel values should definitely not be floats, but uint8
- [ ] enough space should be reserved

For the number of channels, we could use one channel per rgba instead and fill it with the respective value of each pixel?

Reply:
For the channel values, keep in mind that there are also Cameras with HDR and float images. btw: using float in the channel is directly derived form the ROS Pointcloud type: http://docs.ros.org/en/noetic/api/sensor_msgs/html/msg/PointCloud.html